Wikipedia/Code/SearchViewController.swift

```diff
@@ -646,6 +648,50 @@ class SearchViewController: ThemeableViewController, WMFNavigationBarConfiguring
 
 }
 
+extension SearchViewController: UITextViewDelegate {
+    func tappedLink(_ url: URL, sourceTextView: UITextView) {
+        guard let url = URL(string: url.absoluteString) else {
+            return
+        }
+        
+        let legacyNavigateAction = { [weak self] in
+            guard let self else { return }
+            let userInfo: [AnyHashable : Any] = [RoutingUserInfoKeys.source: RoutingUserInfoSourceValue.talkPage.rawValue]
+            navigate(to: url.absoluteURL, userInfo: userInfo)
+        }
+        
+        // first try to navigate using LinkCoordinator. If it fails, use the legacy approach.
+        let navController = customArticleCoordinatorNavigationController
+            ?? self.navigationController
+            ?? self.parent?.navigationController
+
+        if let navController {
+            let linkCoordinator = LinkCoordinator(navigationController: navController, url: url.absoluteURL, dataStore: nil, theme: theme, articleSource: .undefined)
```

Comment:
Just checking, I notice tapping a DYK link opens in the old tab instead of the new tab that the user just tried to create:

https://github.com/user-attachments/assets/a822f54c-ea90-45cd-b0f4-abcb0d0b4917

Is that expected? If not might want to try passing in a different tabConfig value in this LinkCoordinator init.